src/main/java/com/mraof/minestuck/entity/animation/PhasedMobAnimation.java

```diff
@@ -0,0 +1,147 @@
+package com.mraof.minestuck.entity.animation;
+
+import net.minecraft.world.entity.PathfinderMob;
+
+/**
+ * A supplement to MobAnimation that allows for certain code to be performed at the point where one phase of an animation ends and the other begins.
+ * Primarily intended for use with {@link com.mraof.minestuck.entity.ai.MobAnimationPhaseGoal}
+ */
+public class PhasedMobAnimation extends MobAnimation
```

Comment:
Looks like a bad usecase of inheritance. Could you use composition instead, i.e. a `MobAnimation` field with a getter instead of extending?

Also while not strictly neccessary, `MobAnimation` and `PhasedMobAnimation` would be very suitable to be converted to records afterwards.

Reply:
IntelliJ is warning that converting to a Record will weaken visibility of private fields, would this be a problem?

Reply:
Shouldn't be considering that the fields already have getters.